### PR TITLE
Check ironic deployment instead of pods after pivot

### LINKED ISSUE #981: Ironic health is not checked properly after pivot
After pivoting Ironic to the workload cluster, we check that "all pods are running" in this way:
https://github.com/metal3-io/metal3-dev-env/blob/481eae31ed28518668ebf21f7d6e7ebd8ca0e63d/vm-setup/roles/v1aX_integration_test/tasks/move.yml#L155-L166

Unfortunately this does not properly identify issues since it will pass as soon as all pods are (for a split second) in running state. If some pod start crash-looping it will still every now and then get to a running state before it crashes, which means that the test passes even when it should not.

### PR DESCRIPTION
**Pull request by mboukhalfa**
fix #981

### CODE REVIEW

**Comment by mboukhalfa** [comment]
/assign @kashifest 
cc @lentzi90

**Review by lentzi90 — COMMENTED** [review]
This looks good to me!
Can you add also a check for BMO? I think we should check both Ironic and BMO deployments. The old check was for all pods in the cluster but I think the main point here is to see that Ironic and BMO are healthy.

**Comment by kashifest** [comment]
I would rather keep the pods check, it was done for multiple reason, one particular reason being, the provider components needs to be up and running before the pivot operation, you can add the deployment check in addition but do it in a loop for ironic and bmo rather than adding two different tasks.

**Comment by lentzi90** [comment]
> I would rather keep the pods check, it was done for multiple reason, one particular reason being, the provider components needs to be up and running before the pivot operation, you can add the deployment check in addition but do it in a loop for ironic and bmo rather than adding two different tasks.

I guess the question then is if we should check the provider components also with the deployment check? As we have seen with Ironic, the "pods are running" check does not capture crash-looping pods in a good way.

**Comment by kashifest** [comment]
> > I would rather keep the pods check, it was done for multiple reason, one particular reason being, the provider components needs to be up and running before the pivot operation, you can add the deployment check in addition but do it in a loop for ironic and bmo rather than adding two different tasks.
> 
> I guess the question then is if we should check the provider components also with the deployment check? As we have seen with Ironic, the "pods are running" check does not capture crash-looping pods in a good way.

Yeah thats what I meant

File: vm-setup/roles/v1aX_integration_test/tasks/move.yml [hunk 1]
```diff
@@ -165,6 +165,16 @@
     until: (target_running_pods is succeeded) and
            (target_running_pods.resources | length == 0)
 
+  - name: Check if all deployments replicas are available
+    kubernetes.core.k8s_info:
+      kind: Deployment
+      kubeconfig: "/tmp/kubeconfig-{{ CLUSTER_NAME }}.yaml"
+    register: ironic_bmo_deployments
```
Comment (lentzi90) [inline vm-setup/roles/v1aX_integration_test/tasks/move.yml:172]: This is all deployment, right?
```suggestion
    register: deployments
```

**Review by lentzi90 — COMMENTED** [review]
The check looks good to me but it is quite spammy in the logs. Would it be possible to reduce the output somehow?

**Review by lentzi90 — CHANGES_REQUESTED** [review]
The functionality is good, but it is still way too spammy I think.
Check the logs at "Check if pods in running state" and "Check if all deployments replicas are available". It is huge!

**Comment by mboukhalfa** [comment]
> Check if all deployments replicas are available

it is printing some warnings when could not find the unavailable replicas he warning print the full deployment yaml. I can replace the full yaml with the deployment name or omit completely the warning but the final out put it also print all deployments yaml and that is how our ansible verbosity is configured do you want to reduce that also ?

**Comment by lentzi90** [comment]
> > Check if all deployments replicas are available
> 
> it is printing some warnings when could not find the unavailable replicas the warning print the full deployment yaml. I can replace the full yaml with the deployment name or omit completely the warning but the final out put it also print all deployments yaml and that is how our ansible verbosity is configured do you want to reduce that also ?

I would say getting rid of the warning is the most important part. The normal print is ok.

**Review by fmuyassarov — APPROVED** [review]

**Pull request merged**